contract/vault/src/lib.rs

```diff
@@ -0,0 +1,1505 @@
+            "Timelock change already pending"
+        );
+        assert!(
+            (MIN_TIMELOCK_NS..=MAX_TIMELOCK_NS).contains(&as_nanos),
+            "Timelock out of bounds"
+        );
+        if as_nanos > self.timelock_ns {
+            self.timelock_ns = as_nanos;
+            Event::TimelockSet {
+                seconds: new_timelock_secs,
+            }
+            .emit();
+        } else {
+            let valid_at = env::block_timestamp() + self.timelock_ns;
+            self.pending_timelock = Some(PendingValue {
+                value: as_nanos,
+                valid_at,
+            });
+            Event::TimelockChangeSubmitted {
+                new_seconds: new_timelock_secs,
+                valid_at,
+            }
+            .emit();
+        }
+    }
+
+    /// Accepts a pending timelock change after it becomes valid.
+    pub fn accept_timelock(&mut self) {
+        Self::require_owner();
+        if let Some(p) = &self.pending_timelock {
+            assert!(
+                env::block_timestamp() >= p.valid_at,
+                "Timelock not elapsed yet"
+            );
+            self.timelock_ns = p.value;
+            self.pending_timelock = None;
+        } else {
+            env::panic_str("No pending timelock change");
+        }
+    }
+
+    /// Revokes any pending timelock change.
+    pub fn revoke_pending_timelock(&mut self) {
+        Self::assert_guardian_or_owner();
+        self.pending_timelock = None;
+        Event::PendingTimelockRevoked {}.emit();
+    }
+
+    /* ----- Market config / queues ----- */
+    /// Submits a change to a market's supply cap.
+    /// Decreases apply immediately; increases are subject to the governance timelock.
+    #[payable]
+    pub fn submit_cap(&mut self, market: AccountId, new_cap: U128) {
+        Self::assert_curator_or_owner();
+        self.ensure_idle();
+
+        let mut required_deposit: u128 = 0;
+        if self.config.get(&market).is_none() {
+            required_deposit = required_deposit.saturating_add(yocto_for_new_market());
+        }
+        let current_cap = self.config.get(&market).map_or(0, |c| c.cap);
+        if new_cap.0 > current_cap {
+            required_deposit = required_deposit.saturating_add(yocto_for_pending_cap());
+        }
+        require_attached_at_least(required_deposit, "submit_cap");
+
+        let config = match self.config.get_mut(&market) {
+            None => {
+                self.config
+                    .insert(market.clone(), MarketConfiguration::default());
+                Event::MarketCreated {
+                    market: market.clone(),
+                }
+                .emit();
+                // Pre-allocate a market_supply record (principal=0) so allocations don't create storage later
+                self.market_supply.insert(market.clone(), 0);
+                #[allow(clippy::unwrap_used, reason = "No side effects")]
+                self.config.get_mut(&market).unwrap()
```

Comment:
```suggestion
                self.config.get_mut(&market).unwrap_or_else(|e| env::panic_str(&e.to_string()))
```

Should save on debug symbols